operator-framework/src/main/java/com/github/containersolutions/operator/EventScheduler.java

```diff
@@ -0,0 +1,82 @@
+package com.github.containersolutions.operator;
+
+
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watcher;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class EventScheduler<R extends CustomResource> implements Watcher<R> {
+
+    private final static Logger log = LoggerFactory.getLogger(EventDispatcher.class);
+
+    // ConcurrentHashMap instead for locking at hashmap bucket level?
+    private Map<String, Pair<Action, CustomResource>> customEventQueue = Collections.synchronizedMap(new HashMap<>());
+
+    private EventDispatcher eventDispatcher;
+
+    private Integer BACKOFF = 5;
+    private Integer DELAY = 0;
+
+    public <R extends CustomResource> EventScheduler(EventDispatcher<R> eventDispatcher) {
+        this.eventDispatcher = eventDispatcher;
+        startRetryingQueue();
```

Comment:
Let's have this called by the class that creates the Scheduler explicitly. Creating threads in a constructor is not nice, because it's somewhat unexpected. At least I only expect simple initialization logic to happen in a constructor.

Reply:
Added it to the operator instead & adapted the tests. 
I can still squash my many commits if require